File: HoneydewTestProject/TrainWreck.cs
namespace HoneydewTestProject
{
    internal class TrainWreck
    {
        public void Method()
        {
            var x = new Wagon4().Method4().Method3().Method2().X;
        }
    }

    public class Wagon2
    {
        public Wagon1 Method2()
        {
            return new Wagon1();
        }
    }

    public class Wagon3
    {
        public Wagon2 Method3()
        {
            return new Wagon2();
        }
    }

    public class Wagon4
    {
        public Wagon3 Method4()
        {
            return new Wagon3();
        }
    }

    public class Wagon1
    {
        public int X { get; set; }
    }
}

File: vb/HoneydewTestProjectVB/HoneydewTestProjectVB/TrainWreck.vb
    Friend Class TrainWreck
        Public Sub Method()
            Dim x = New Wagon4().Method4().Method3().Method2().X
        End Sub
    End Class

    Public Class Wagon2
        Public Function Method2() As Wagon1
            Return New Wagon1()
        End Function
    End Class

    Public Class Wagon3
        Public Function Method3() As Wagon2
            Return New Wagon2()
        End Function
    End Class

    Public Class Wagon4
        Public Function Method4() As Wagon3
            Return New Wagon3()
        End Function
    End Class

    Public Class Wagon1
        Public Property X As Integer
    End Class
